### PR TITLE
* Implement StartTime parsing from Movements Page for Fleet struct

### PR DESCRIPTION
**Pull request by 0xE232FE**
I found it usefull to have the StartTime of a fleet. It help me for many things and for logging perpose.

The StartTime for not returning Fleet is the Timestamp when the Fleet has been dispatched.
The StartTime for returning Fleet is the Timestamp of the Recall time when you recalled the fleet.

The Timestamp is adjusted by the Timezone because the HTML Timestamp doesn't contain any timezone and is OGame Time specific so I take the help of the ExtractOGameTimestamp.

### CODE REVIEW

**Comment by 0xE232FE** [comment]
Now its working with any Language.

**Comment by flashover** [comment]
Any chance for a testcase?

**Comment by flashover** [comment]
You have two if-blocks. In the outer if-block these are the only line that changes:
`startTimeString, _ := s.Find("div.origin img").Attr("title")`
and:
`startTimeString, _ := s.Find("div.destination img").Attr("title")`

Maybe you can simplify the code as the other lines are exactly the same and move those outside the outer if-block?

**Comment by 0xE232FE** [comment]
@flashover: Agreed to the simplification and removed repeating code.
Test-case is coming.

File: extracts_v6.go [hunk 1]
```diff
@@ -1167,6 +1169,17 @@ func extractFleetsFromDocV6(doc *goquery.Document, clock clockwork.Clock) (res [
 			secs = 0
 		}
 
+
+		var startTimeString string
+		if !returnFlight {
+			startTimeString, _ = s.Find("div.origin img").Attr("title")
+		} else {
+			startTimeString, _ = s.Find("div.destination img").Attr("title")
+		}
```
Comment (flashover) [inline extracts_v6.go]:
```suggestion
		if !returnFlight {
			startTimeString, startTimeStringErr = s.Find("div.origin img").Attr("title")
		} else {
			startTimeString, startTimeStringErr = s.Find("div.destination img").Attr("title")
		}
```

File: extracts_v6.go [hunk 2]
```diff
@@ -1167,6 +1169,17 @@ func extractFleetsFromDocV6(doc *goquery.Document, clock clockwork.Clock) (res [
...
+		startTimeArray := strings.Split(startTimeString, ":| ")
+		startTime, _ := time.ParseInLocation("02.01.2006<br>15:04:05", startTimeArray[1], servertime.Location())
```
Comment (flashover) [inline extracts_v6.go]: Maybe for safety we need to check if startTimeArray[1] exists with len() == 2? :

```suggestion
	if !startTimeStringErr {
		startTimeArray := strings.Split(startTimeString, ":| ")
		if len(startTimeArray) == 2 {
			startTime, _ := time.ParseInLocation("02.01.2006<br>15:04:05", startTimeArray[1], servertime.Location())
		}
	}
```

**Comment by alaingilbert** [comment]
According to my tests, the start time for the recall fleet seems to be the real start time... isn't it ?

```html
<div class="origin fixed">
--
  | <img class="tooltipHTML" height="30" width="30" src="https://gf1.geo.gfsrv.net/cdn9e/6183cba4025fe240b2d2fe7dc84f92.png" title="Start time:\| 16.03.2020&lt;br&gt;05:58:46" alt=""/>
  | </div>
```
```html
<div class="destination fixed">
--
  | <img class="tooltipHTML" height="30" width="30" src="https://gf1.geo.gfsrv.net/cdn07/66f44c7a5a76f653320c621afcd0c7.png" title="Start time:\| 16.03.2020&lt;br&gt;05:58:46" alt=""/>
  | </div>
```

**Pull request merged**